dynamic/internal/fixup/properties.go

```diff
@@ -100,54 +101,102 @@ func fixPropertyConflict(p *info.Provider) error {
 		return fmt.Errorf("must set p.Name")
 	}
 	return walkResources(p, func(r tfbridge.Resource) error {
-		var retError error
+		var retError []error
 		r.TF.Schema().Range(func(key string, value shim.Schema) bool {
 			if fix := badPropertyName(p.Name, key); fix != nil {
-				if r.Schema.Fields == nil {
-					r.Schema.Fields = map[string]*info.Schema{}
-				}
-
-				s, ok := r.Schema.Fields[key]
-				if !ok {
-					s = &info.Schema{}
-				}
-
-				if s.Name == "" {
-					var err error
-					s.Name, err = fix(r.TF)
-					if err != nil {
-						retError = fmt.Errorf("%q: %w", key, err)
-					}
-				}
-
-				if !ok {
-					r.Schema.Fields[key] = s
+				err := fix(r)
+				if err != nil {
+					retError = append(retError, fmt.Errorf("%s: %w", key, err))
 				}
 			}
 
 			return true
 		})
-		return retError
+		return errors.Join(retError...)
 	})
 }
 
-func badPropertyName(providerName, key string) func(shim.Resource) (string, error) {
+func getField(i *map[string]*info.Schema, name string) *info.Schema {
+	contract.Assertf(i != nil, "i cannot be nil")
+	if *i == nil {
+		*i = map[string]*info.Schema{}
+	}
+	v, ok := (*i)[name]
+	if !ok {
+		v = new(info.Schema)
+		(*i)[name] = v
+	}
+	return v
+}
+
+type fixupProperty = func(tfbridge.Resource) error
+
+func badPropertyName(providerName, key string) fixupProperty {
 	switch key {
 	case "urn":
-		return newName(providerName)
+		return fixURN(providerName)
+	case "id":
+		return fixID(providerName)
 	default:
 		return nil
 	}
 }
 
-func newName(providerName string) func(shim.Resource) (string, error) {
-	return func(r shim.Resource) (string, error) {
-		s := r.Schema()
+func fixID(providerName string) fixupProperty {
+	return func(r tfbridge.Resource) error {
+		s := r.TF.Schema()
+		v := providerName + "_id"
```

Comment:
Can we please add some meaningful variable names here? This could be `proposedNewName`, `resourceSchema` or `resSchema` above. `id` is overloaded, so could be `idProp`, etc...

Reply:
Are we renaming the `id` field to `providerName_id`? Should we instead rename it to something more generic like `idField` or something like that?

Reply:
We are renaming the `"id"` to `"provider_name_id"` (`"providerNameId"` when projected into Pulumi). I think that `"idField"` is more confusing. At least `"providerNameId"` indicates that it's the ID field associated with the provider.

Reply:
I would say that `providerNameId` is pretty generic in the context of `providerName`. It's also practically guaranteed to be generic, where `idField` could be interpreted as associated with some "field" on the upstream cloud resource itself.

Reply:
Are we renaming the "id" field on a resource here? I feel like it might be a bit weird with some providers if the field contains the provider name - what was the provider with a `-` from last week? How does this look with some of the longer provider names?

Reply:
> Are we renaming the "id" field on a resource here?

Yes.

>  I feel like it might be a bit weird with some providers if the field contains the provider name - what was the provider with a - from last week? How does this look with some of the longer provider names?

 We are taking the unaltered TF name and running it through `TerraformToPulumiNameV2`. It should be handled the same way as other fields. I'll validate the case of a kabob-case provider name.

Reply:
@iwahbe @VenelinMartinov @t0yv0 If I could have wished for anything, it would have gone for the field to be called `<resourceName>Id`. My code example in #2383 would end up now with:

```python
_ = DeploymentType(
  resource_name = "workloads-v1",
  uicloudprovider_id = "workloads.v1",
  display_name = "Workload",
  ...
)
```

To me this is very confusing because it might indicate that the field excepts the id of the provider instance in some way.

If the field name would be `deploymentType_id` the intend is obvious.

```python
_ = DeploymentType(
  resource_name = "workloads-v1",
  deploymentType_id = "workloads.v1",
  display_name = "Workload",
  ...
)
```

Reply:
I think the `provider` part is dropped and the field should actually be:
```python
_ = DeploymentType(
  resource_name = "workloads-v1",
  uicloud_id = "workloads.v1",
  display_name = "Workload",
  ...
)
```

Do you think that's confusing?

Reply:
@VenelinMartinov yes of course, because everyone would ask "what cloud id? Do I've to get it from somewhere else?". There is no obvious reference to the fact that you simply have to specify the (unique) ID of the (new) `DeploymentType` resource. What in case of the `DeploymentType` resource is a simple string that the practitioner can choose arbitrarily. The back-end will create a `409` if the type already exists though.

Reply:
Actually, for the provider you are trying to bridge, is it called uicloud or uicloudprovider? If it's public, can you share it with us?

Reply:
@VenelinMartinov  Unable to share because it's an internal provider in a customer project. So at first glance I'd say I won't get the permission to share it. Current (file-)name is `terraform-provider-uicloudplatform`. So the provider name is `uicloudprovider`.

Reply:
@iwahbe opened https://github.com/pulumi/pulumi-terraform-bridge/pull/2394 to address this. Thank you for the feedback!

Reply:
@iwahbe @VenelinMartinov YAY! 🥳 Thanks for that! Much appreciated. :+1